_posts/2020-09-06-how-the-system-tests-works.md

```diff
@@ -0,0 +1,599 @@
+---
+layout: post
+title:  "How system tests works"
+date: 2020-09-06
+author: maroš_orsák
+---
+
+Sometimes, we underestimate strength of testing. Many companies start verification of the software product belated.
+The process of testing is complex as many can not imagine. 
+It can be highlighted by quote: "If we fail, we fall. If we succeed - then we will face the next task." 
+Fortunately, Strimzi is an exception but you have to know many technologies such as [Kubernetes](https://kubernetes.io/), 
+[Apache Kafka](https://kafka.apache.org/),[Keycloak](https://www.keycloak.org/), [Open Policy Agent](https://www.openpolicyagent.org/),
+[Cruise control](https://github.com/linkedin/cruise-control) and so on.
+
+In this blog post we are gonna take a closer look on our system tests. The content is as follows:
+
+# Content
+1. [Introduction](#Introduction)
+2. [Resources](#Resources)
+3. [Lifecycle of tests](#Lifecycle of tests)
+4. [Auxiliary classes](#Auxiliary classes)
+5. [Dependencies](#Dependencies)
+6. [How to create a system test](#How to create a system test)
+8. [Conclusion](#Conclusion)
+
+## Introduction
+
+The whole ecosystem behind systems test is encapsulated in two mandatory things. First of all it is
+the Resources, which create the whole testing environment. The second of all we have the auxiliary classes that are divided into
+the classic static methods `Utils`, internal clients and also `Apache Kafka Clients` for the external communication, 
+Kubernetes client, `Constants.java` and `Environment.java`.
+
+Pure idea to make the system tests easily modifiable is to able to write resources in the fluent way. 
+Fluent notation are nested builders by which you can build customized objects by your imagination.
+For instance before the Fluent notation we have the [1.Telescoping pattern](https://www.vojtechruzicka.com/avoid-telescoping-constructor-pattern/#telescoping-constructor) 
+but this does not scale well for instance in situation with large numbers of optional parameters. 
+Second alternative would be the [2.JavaBeans convention](https://www.vojtechruzicka.com/avoid-telescoping-constructor-pattern/#alternative-1---javabeans), 
+which first of all resolve our issue from the telescoping pattern but on the other hand the objects which we create will be mutable. 
+Luckily, there is a third alternative that combines the safety of the telescoping constructor pattern with the readability 
+of the `JavaBeans pattern`. 
+It is a form of the [3.Builder pattern](https://refactoring.guru/design-patterns/builder).
+Moreover, if we take this idea to the next level to the `4.Abstract Recursive builder` we will be able to create multiple 
+dependent objects, which have nested hierarchy. In the following code snippets 1.1, 1.2, 1.3 we can see described idea and you 
```

Comment:
I think that you forgot to change the snippets "names" down there. Also is it possible to add some reference to the snippet? Just an idea..